RELEASING.md

```diff
@@ -0,0 +1,264 @@
+# Versioning and release processes
+
+## Versioning Process
+
+Ledger team has fully adopted the [PVP versioning scheme](https://pvp.haskell.org/) for
+all packages in this repo.
+
+Updating the version for every package in this repository is the responsibility of the
+developer that is introducing a change to any package that is subject to release
+process. Version must be updated in both the `CHANGELOG.md` and in the `.cabal` file for
+the package in the same PR, as well as every other package that is affected by the change
+in the repository. Every affected package that is subject to release process must receive
+a lower and, potentially, an upper bound update on the dependency that has experienced the
+change.
+
+For example if I add a function to `cardano-ledger-core` and use it in
+`cardano-ledger-conway`, then I must add a minor version bump to `cardano-ledger-core` and
+I also must add a lower version restriction in `cardano-ledger-conway` for
+`cardano-ledger-core`, as per PVP.
+
+In other words, when it comes to versioning, the fact that all of the packages in this
+repository are built together at the same time should be ignored. It should be assumed
+that every package and all of its required dependencies can be released to CHaPs at any
+point in time. This is necessary to guarantee that if such a package is compatible with
+other packages from this repository, which have been previously released, the version
+bounds should reflect that.
+
+### Upper bounds
+
+Normally, [upper bounds are required by
+PVP](https://pvp.haskell.org/#dependencies-in-cabal) for all dependencies of a
+package. This is the only exception we make to the PVP, and we do not require strict upper
+bounds for dependencies that come from Hackage. For example, there is no need to add an
+upper bound on `containers` package, unless there is a known incompatibility. Speculative
+upper bounds over time lead to overly restrictive build plans that are forced to favor
+older packages. Moreover, they lead to a lot of redundant maintenance burden. That being
+said, most of the packages from within the same repository are a lot more volatile and
+require tight interoperability, therefore speculative upper bounds are encouraged.
```

Comment:
We are taking about the ledger-state repository when it says "within the same repository" right?
So might it be better to say,  we encourage upper bounds on packages within the ledger-state repository.

Reply:
Yeah, you are right, I'll reword it to be more ledger specific.

Reply:
Done